### PR TITLE
TEP-0021 Result API: Record Subresources

### PR DESCRIPTION
**Pull request by wlynch**
- Renames Event -> Record based on feedback (still has the same intent)
- Breaks Records out as a subresource of Results to allow for pagination
  and finer grained access.

/cc @afrittoli

### CODE REVIEW

**Review by sbwsg — APPROVED** [review]

File: teps/0021-results-api.md [hunk 1]
```diff
@@ -457,11 +457,6 @@ message Result {
...
   map<string, string> annotations = 3;
 
-  // Execution artifacts part of this result. Typically will be Tekton
-  // Task/PipelineRuns, but may also include other execution information
-  // (e.g. DSLs, etc.)
-  repeated Event events = 4;
-
   // The etag for this result.
   // If this is provided on update, it must match the server's etag.
   string etag = 6
```
Comment (afrittoli) [inline teps/0021-results-api.md:462]: NIT: should this become #4 now?

**Review by afrittoli — COMMENTED** [review]
Thanks for the update!
/approve

File: teps/0021-results-api.md (hunk 1 above)
Comment (wlynch) [reply teps/0021-results-api.md:462]: Technically for protos you should never reuse field numbers and just mark them as `reserved`. For proposal docs, this doesn't matter too much - I'll be reindexing these once this proto becomes real.

**Pull request merged**